searchcore/src/vespa/searchcore/proton/server/executorthreadingservice.cpp

```diff
@@ -21,12 +22,16 @@ namespace proton {
 namespace {
 
 std::unique_ptr<SyncableThreadExecutor>
-createExecutorWithOneThread(uint32_t taskLimit, OptimizeFor optimize,
-                            vespalib::Runnable::init_fun_t init_function) {
-    if (optimize == OptimizeFor::THROUGHPUT) {
-        return std::make_unique<SingleExecutor>(std::move(init_function), taskLimit);
+createExecutorWithOneThread(const ThreadingServiceConfig & cfg, vespalib::Runnable::init_fun_t init_function) {
+    uint32_t taskLimit = cfg.defaultTaskLimit();
+    if (cfg.optimize() == OptimizeFor::THROUGHPUT) {
+        return std::make_unique<SingleExecutor>(std::move(init_function), taskLimit, cfg.is_task_limit_hard(), taskLimit/10, 100ms);
```

Comment:
Consider adding a constructor for SingleExecutor taking 3 parameters to avoid repeating default values (taskLimit/10, 100ms) here.

Reply:
I removed the default constructor instead, and updated the test that used it. That was the reason why this was not caught when we were working with this last time.